daemon/SecretsImpl/secrets.cpp

```diff
@@ -1107,14 +1133,14 @@ void Daemon::ApiImpl::SecretsRequestQueue::handleFinishedRequest(
                 qCWarning(lcSailfishSecretsDaemon) << "UserInputRequest:" << request->requestId << "finished as pending!";
                 *completed = true;
             } else {
+                const QByteArray userInput = request->outParams.size()
+                        ? request->outParams.takeFirst().value<QByteArray>()
+                        : QByteArray();
                 if (request->isSecretsCryptoRequest) {
-                    QByteArray userInput = request->outParams.size()
-                            ? request->outParams.takeFirst().value<QByteArray>()
-                            : QByteArray();
                     asynchronousCryptoRequestCompleted(request->cryptoRequestId, result, QVariantList() << userInput);
                 } else {
-                    // shouldn't happen!
-                    qCWarning(lcSailfishSecretsDaemon) << "Error: UserInputRequest not a secrets crypto request!";
+                    request->connection.send(request->message.createReply() << QVariant::fromValue<Result>(result)
```

Comment:
Interesting, it was "should not happen" earlier and now it is okay? Needs some explanation.

Reply:
Because before, the only codepath the UserInputRequest was used in, was as an asynchronous crypto request (i.e. when getting input key data for a key derivation request).  Now it is used also for "standalone" secrets request (i.e. InteractionRequest).